app/services/parity/ParityExportService.scala

```diff
@@ -0,0 +1,127 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.parity
+
+import java.util.UUID
+
+import config.{ MicroserviceAppConfig, ParityGatewayConfig }
+import connectors.paritygateway.ParityGatewayClient
+import model.EvaluationResults.Green
+import model.ProgressStatuses.EXPORTED
+import play.api.Logger
+import play.api.libs.json._
+import services.events.{ EventService, EventSink }
+import repositories._
+import repositories.application.GeneralApplicationRepository
+import repositories.parity.{ ApplicationReadyForExport, ParityExportRepository }
+import repositories.parity.ParityExportRepository.ApplicationIdNotFoundException
+import services.onlinetesting.EvaluatePhase3ResultService
+import services.reporting.{ SocioEconomicCalculator, SocioEconomicScoreCalculator }
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object ParityExportService extends ParityExportService {
+  val eventService = EventService
+  val parityExRepository = parityExportRepository
+  val parityGatewayConfig = MicroserviceAppConfig.parityGatewayConfig
+  val parityGatewayClient = ParityGatewayClient
+  val mRepository = mediaRepository
+  val cdRepository = faststreamContactDetailsRepository
+  val qRepository = questionnaireRepository
+  val northSouthRepository = northSouthIndicatorRepository
+  val evaluateP3ResultService = EvaluatePhase3ResultService
+  val socioEconomicCalculator = SocioEconomicCalculator
+  val appRepository = applicationRepository
+}
+
+trait ParityExportService extends EventSink {
+
+  val parityExRepository: ParityExportRepository
+  val parityGatewayConfig: ParityGatewayConfig
+  val parityGatewayClient: ParityGatewayClient
+  val mRepository: MediaRepository
+  val cdRepository: contactdetails.ContactDetailsRepository
+  val qRepository: QuestionnaireRepository
+  val northSouthRepository: NorthSouthIndicatorCSVRepository
+  val evaluateP3ResultService: EvaluatePhase3ResultService
+  val socioEconomicCalculator: SocioEconomicScoreCalculator
+  val appRepository: GeneralApplicationRepository
+
+  // Random apps in READY_FOR_EXPORT
+  def nextApplicationsForExport(batchSize: Int): Future[List[ApplicationReadyForExport]] =
+  parityExRepository.nextApplicationsForExport(batchSize)
+
+  def exportApplication(applicationId: String): Future[Unit] = {
+    for {
+      exportJson <- generateExportJson(applicationId)
+      _ <- parityGatewayClient.createExport(exportJson)
+      _ <- appRepository.addProgressStatusAndUpdateAppStatus(applicationId, EXPORTED)
+    } yield ()
+  }
+
+  // scalastyle:off method.length
+  private def generateExportJson(applicationId: String): Future[JsObject] = {
```

Comment:
(Suggestion only) Could you extract parts of this into methods/functions to make it a bit more intention revealing?